vpr/src/route/partition_tree.cpp

```diff
@@ -10,19 +11,24 @@ constexpr size_t MIN_NETS_TO_PARTITION = 256;
 PartitionTree::PartitionTree(const Netlist<>& netlist) {
     const auto& device_ctx = g_vpr_ctx.device();
 
-    auto all_nets = std::vector<ParentNetId>(netlist.nets().begin(), netlist.nets().end());
+    auto all_nets = std::unordered_set<ParentNetId>(netlist.nets().begin(), netlist.nets().end());
     _root = build_helper(netlist, all_nets, 0, 0, device_ctx.grid.width() - 1, device_ctx.grid.height() - 1);
 }
 
-std::unique_ptr<PartitionTreeNode> PartitionTree::build_helper(const Netlist<>& netlist, const std::vector<ParentNetId>& nets, int x1, int y1, int x2, int y2) {
+std::unique_ptr<PartitionTreeNode> PartitionTree::build_helper(const Netlist<>& netlist, const std::unordered_set<ParentNetId>& nets, int x1, int y1, int x2, int y2) {
     if (nets.empty())
         return nullptr;
 
     const auto& route_ctx = g_vpr_ctx.routing();
     auto out = std::make_unique<PartitionTreeNode>();
 
     if (nets.size() < MIN_NETS_TO_PARTITION) {
+        out->bb = {x1, x2, y1, y2, 0, 0};
```

Comment:
Should either make this work for multi-layer FPGAs, or throw an error at command line parsing or in some set up routine if it is a multi-layer FPGA and a parallel router is chosen.

Try testing it for a 3D case ... if it works that's fine.  I'm not worried about performance tuning it.